tests/utils/test_config.py

```diff
@@ -93,3 +94,61 @@ def test_from_configuration_files_get_typed_value(tmp_path_factory: pytest.TempP
 
     assert Config().get_bool("legacy-current-snapshot-id")
     assert Config().get_int("max-workers") == 4
+
+
+@pytest.mark.parametrize(
+    "config_location, config_content, expected_result",
+    [
+        (
+            "config",
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+        ),
+        (
+            "home",
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+        ),
+        (
+            "current",
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+            {"catalog": {"default": {"uri": "https://service.io/api"}}},
+        ),
+        ("none", None, None),
+    ],
+)
+def test_from_multiple_configuration_files(
```

Comment:
nit can we also test the order of lookup here?

Reply:
we want to test both home directory and current directory here

Reply:
I have just pushed an adjustment in https://github.com/apache/iceberg-python/pull/1464/commits/fe029fe40340186363bd4031ebb07a2219336d72 that aims to resolve this. Thank you for the feedback!